### PR TITLE
Add staircase

### PR DESCRIPTION
**Pull request by venaturum**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/staircase/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: ded45fb4444bb5273cee6ca13d092952a908c72968eef0c26776996adae4a1ec
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vv
```
Comment (synapticarbors) [inline recipes/staircase/meta.yaml:16]: The current recommended `script` command is:

```
script: "{{ PYTHON }} -m pip install . -vv"
```

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L35

File: recipes/staircase/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - sortedcontainers >=2,<3
+
+test:
+  imports:
+    - staircase
+    - staircase.docstrings
+  source_files:
+    - tests
+  requires:
+    - python
```
Comment (synapticarbors) [inline recipes/staircase/meta.yaml:36]: You don't have to list `python` under `requires` since it is pulled in as a `run` requirement.

**Comment by venaturum** [comment]
Thanks @synapticarbors, think I was led astray by a few other recipes I found around the traps.

Have rectified in latest commit.  Much appreciated.

**Comment by venaturum** [comment]
@conda-forge/staged-recipes

I believe this is ready to go if you have time to check.  Thank you.

**Pull request closed without merging**